utils/navigation_decorator.py

```diff
@@ -0,0 +1,12 @@
+import cfme.fixtures.pytest_selenium as sel
+import cfme.web_ui.menu as menu
+from method_decorator import method_decorator
+
+
+def navigate(page_name):
```

Comment:
I believe it should rather be a function with proper wrapping, but yes, I agree something like that would be convenient. Then if we implement some kind of detector, then it can also check - do I have any detector on this page name? if not, navigate, otherwise check if I am on the page and if I am not then navigate. And we don't have to write all these checks inside the methods ...